tests/scripts/thread-cert/Cert_5_2_05_AddressQuery.py

```diff
@@ -30,110 +30,130 @@
         self.nodes[LEADER].set_state('leader')
         self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+        self.nodes[LEADER].add_prefix('2001::/64', 'pdros')
+        self.nodes[LEADER].register_netdata()
 
-        self.nodes[BR1].start()
+        # 2. BR: Attach the Border_Router to the network and configure On-Mesh Prefix.
+        self.nodes[BR].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[BR1].get_state(), 'router')
-
-        self.nodes[BR1].add_prefix('2001:2:0:3::/64', 'paros')
-        self.nodes[BR1].add_prefix('2001:2:0:4::/64', 'paros')
-        self.nodes[BR1].register_netdata()
+        self.assertEqual(self.nodes[BR].get_state(), 'router')
+        self.nodes[BR].add_prefix('2002::/64', 'paros')
+        self.nodes[BR].register_netdata()
+
+        # 3. All: Ensure topology is formed correctly without the DUT_REED.
+        for i in range(2, 17):
+            if i == BR:
+                continue
+            self.nodes[i].start()
+            time.sleep(5)
+            self.assertEqual(self.nodes[i].get_state(), 'router')
 
         self.nodes[ED1].start()
         time.sleep(5)
         self.assertEqual(self.nodes[ED1].get_state(), 'child')
 
-        self.nodes[REED].start()
+        # 4. DUT_REED: Cause the DUT_REED to attach to ROUTER1.
+        self.nodes[DUT_REED].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[REED].get_state(), 'child')
+        time.sleep(ROUTER_SELECTION_JITTER)
 
-        self.nodes[ROUTER2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ROUTER2].get_state(), 'router')
+        reed_messages = self.sniffer.get_messages_sent_by(DUT_REED)
 
-        self.nodes[ED2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED2].get_state(), 'child')
+        # The DUT_REED must not send a coap message here.
+        msg = reed_messages.does_not_contain_coap_message()
+        assert msg is True, "Error: The REED sent an Address Solicit Request"
 
-        self.nodes[ED3].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED3].get_state(), 'child')
+        # 5. Enables a link between the DUT and BR to create a one-way link.
+        self.nodes[DUT_REED].add_whitelist(self.nodes[BR].get_addr64())
+        self.nodes[BR].add_whitelist(self.nodes[DUT_REED].get_addr64())
+
+        # 6. Send an ICMPv6 Echo Request from ED_1 to REED_1 using ML-EID/2001:: EID/2002:: EID to test if the DUT_REED would send a properly formatted Address Notification or not.
+        mleid = self.nodes[DUT_REED].getIp6Address('mesh EID')
+        self.assertTrue(self.nodes[ED1].ping(mleid))
+        self.checkAddressNotification()
 
-        addrs = self.nodes[REED].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[ED2].ping(addr))
-                time.sleep(1)
+        for global_address in self.nodes[DUT_REED].getIp6Address('global'):
```

Comment:
add comment before Line #154 as 
`# 7 & 8. Verify DUT_REED would send Address Notification when ping to its 2001::EID and 2002::EID.`

Reply:
How about verifying DUT_REED has 2001::EID and 2002::EID and ping to them respectively?